feature_tests/c2/include/BorrowedFields.h

```diff
@@ -0,0 +1,34 @@
+#ifndef BorrowedFields_H
+#define BorrowedFields_H
+
+
+#include <stdio.h>
+#include <stdint.h>
+#include <stddef.h>
+#include <stdbool.h>
+#include "diplomat_runtime.h"
+
+
+#ifdef __cplusplus
+namespace capi {
+extern "C" {
+#endif // __cplusplus
+
+
+
+typedef struct BorrowedFields {
+	const uint16_t* a_data;
+	size_t a_len;
```

Comment:
Observation: C2 spits out two separate fields for the slice, but not the string, and C1 uses a combined type for both the slice and the string. Is it intentional? I observe that diplomat_runtime.h is adding the primitive slice wrappers, so should they be used here?

In general, why do we have these wrappers for struct fields but not function arguments?

Reply:
Yeah it's actually incorrect for C2 to spit out two separate fields for the slice (I noticed this yesterday and plan to fix it)

```
#[repr(C)]
struct Foo {
   a: (*const u8, usize)
}
```

is not guaranteed to have the same ABI as

```rust
#[repr(C)]
struct Foo {
  a_ptr: *const u8,
  a_size: usize
}
```

Reply:
Actually, I'm _not sure_ of the above. I think it _might_ be true but when I was first working on it it wasn't fully clear.